### PR TITLE
[TFL] Fix unbounded activation clip range.

### LINKED ISSUE #45312: TFlite gets the incorrect value dividing by zero or computing tf.log(x)
**System information**
- OS Platform and Distribution: MacOS Catalina 10.15.6
- TensorFlow installed: from binary
- TensorFlow version: The issue could be reproduced by TF1.x (TF 1.15.2) and TF2.x (TF 2.3.1)
- Python version: 3.6.5

**Describe the current behavior**
I have the following code that simply creates a TF graph whose output tensor is an input placeholder divided by a float32 constant 0. I would expect the evaluation value of the output tensor to be always "Inf" no matter which value is fed into the input placeholder. However, what I got from the following example is, the result from Tensorflow is expected, while the one from TFlite is the max limit of float32.

```
import os
import re
import tempfile

import numpy as np
import tensorflow as tf

is_tf_2 = bool(re.match("2\.[0-9]+\.[0-9]+", tf.version.VERSION))
if is_tf_2:
    print("using TF2.x")
    import tensorflow.compat.v1 as tf

    tf.compat.v1.disable_eager_execution()


def run_tf_ops():
    input_tensor = tf.placeholder(dtype=tf.float32, shape=[None])
    b = tf.constant(0.0, dtype=tf.float32)
    output_tensor = tf.divide(input_tensor, b)

    tf_session = tf.Session()

    with tempfile.TemporaryDirectory("") as tempdir:
        converter = tf.lite.TFLiteConverter.from_session(
            sess=tf_session,
            input_tensors=[input_tensor],
            output_tensors=[output_tensor],
        )
        tflite_model = converter.convert()
        tflite_saved_model_path = os.path.join(tempdir, "saved_model.tflite")
        with open(tflite_saved_model_path, "wb") as f:
            f.write(tflite_model)

        # Test TFLite load
        # Load the TFLite model and allocate tensors.
        interpreter = tf.lite.Interpreter(model_path=tflite_saved_model_path)
        interpreter.allocate_tensors()
        # Get input and output tensors.
        input_details = interpreter.get_input_details()
        output_details = interpreter.get_output_details()

        # Test the model on random input data.
        # get output from tflite model
        interpreter.set_tensor(input_details[0]["index"], [np.float32(1.0)])
        interpreter.invoke()
        output_data_from_tflite = interpreter.get_tensor(output_details[0]["index"])

    print(
        f"Tensorflow result: {tf_session.run(output_tensor, feed_dict={input_tensor: [np.float32(1.0)]})}"
    )
    print(f"TFLite result: {output_data_from_tflite}")


if __name__ == "__main__":
    run_tf_ops()
```
The interesting behavior is if I replace the input placeholder `input_tensor` with a constant float32 tensor like `input_tensor = tf.constant(value=[1.0], dtype=tf.float32)`(also remove the code on feeding data), both Tensorflow and TFlite get the correct result `Inf` in TF 1.15.2, but would have the same issue in TF 2.3.1.

**NOTE** The same behavior happens for other operations such as "tf.log(x)" when we feed x with run time data 0.

**Standalone code to reproduce the issue**
The issue could be 100% reproduced by running the above code with the system info.

### PR DESCRIPTION
**Pull request by WindQAQ**
Fixes #45312. For floating point with unbounded activation such as relu and none, set max/min to positive/negative infinity in case infinity is clipped. The change only affects floating point as `std::numeric_limits<T>::has_infinity` is always false for non-floating point. Doing `min(max(x, -inf), inf)` is safe because positive infinity is larger than all values except itself and NaN and negative infinity is smaller than all values expect itself and NaN. With the change, TFLite is the same with TF for relu and none activation when input is positive or negative infinity.

https://www.gnu.org/software/libc/manual/html_node/Infinity-and-NaN.html
https://en.cppreference.com/w/cpp/types/numeric_limits/has_infinity

### CODE REVIEW

File: tensorflow/lite/kernels/internal/BUILD [hunk 1]
```diff
@@ -952,6 +952,19 @@ cc_test(
     ],
 )
 
+cc_test(
+    name = "common_test",
+    srcs = [
+        "common_test.cc",
+    ],
+    shard_count = 2,
```
Comment (abattery) [inline tensorflow/lite/kernels/internal/BUILD:960]: nit: we can remove this since the total test execution time will be quite small.

**Review by abattery — APPROVED** [review]

File: tensorflow/lite/kernels/internal/BUILD (hunk 1 above)
Comment (WindQAQ) [reply tensorflow/lite/kernels/internal/BUILD]: Updated 😃

**Review by abattery — APPROVED** [review]

**Review by thaink — APPROVED** [review]

File: tensorflow/lite/kernels/internal/common_test.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                           &activation_min, &activation_max);
+
+  std::transform(input.begin(), input.end(), output.begin(), [&](float x) {
+    return ActivationFunctionWithMinMax(x, activation_min, activation_max);
+  });
+  EXPECT_THAT(output, ElementsAreArray(input));
+}
+
+}  // namespace
+}  // namespace tflite
```
Comment (abattery) [inline tensorflow/lite/kernels/internal/common_test.cc:79]: Could you add a newline character at the end of the file? This blocks an warnining message from a lint pass.

Comment (WindQAQ) [reply tensorflow/lite/kernels/internal/common_test.cc:79]: Updated 😃

**Review by abattery — APPROVED** [review]

File: tensorflow/lite/kernels/internal/common_test.cc (hunk 2 above)
Comment (thaink) [inline tensorflow/lite/kernels/internal/common_test.cc:79]: Can you add a newline at the end of this file?
It failed our lint test.

Comment (WindQAQ) [reply tensorflow/lite/kernels/internal/common_test.cc:79]: I think I've already added it at the latest commit. Am I missing something here?

Comment (abattery) [reply tensorflow/lite/kernels/internal/common_test.cc:79]: Confirmed that this is already resolved in the last commit.

**Review by miaout17 — APPROVED** [review]

File: tensorflow/lite/kernels/internal/common_test.cc (hunk 2 above)
Comment (thaink) [reply tensorflow/lite/kernels/internal/common_test.cc:79]: Acked. Let me try to sync this to our internal PR.

**Review by thaink — APPROVED** [review]

**Comment by thaink** [comment]
This PR will be hold until we can invest to make a thorough investigation about how current ops are handling inf value to ensure there is no compatibility issues.

**Pull request closed without merging**